ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/VersionedOffsetBuilderSpec.scala

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.participant.state.kvutils
+
+import org.scalacheck.Arbitrary.arbitrary
+import org.scalacheck.Gen
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import org.scalatestplus.scalacheck.ScalaCheckDrivenPropertyChecks
+
+/** Other test cases are covered by [[OffsetBuilderSpec]] */
+class VersionedOffsetBuilderSpec
+    extends AnyWordSpec
+    with Matchers
+    with ScalaCheckDrivenPropertyChecks {
+  import VersionedOffsetBuilderSpec._
+
+  "VersionedOffsetBuilder" should {
+    "construct and extract" in {
+      forAll(arbitrary[Byte], genHighest, Gen.posNum[Int], Gen.posNum[Int]) {
+        (version, highest, middle, lowest) =>
+          val builder = VersionedOffsetBuilder(version)
+          val offset = builder.of(highest, middle, lowest)
+
+          builder.version(offset) should be(version)
+          builder.highestIndex(offset) should be(highest)
+          builder.middleIndex(offset) should be(middle)
+          builder.lowestIndex(offset) should be(lowest)
+          builder.split(offset) should be((highest, middle, lowest))
+      }
+    }
+
+    "fail on a highest that is out of range" in {
+      forAll(arbitrary[Byte], genOutOfRangeHighest, Gen.posNum[Int], Gen.posNum[Int]) {
+        (version, highest, middle, lowest) =>
+          the[IllegalArgumentException] thrownBy VersionedOffsetBuilder(version).of(
+            highest,
+            middle,
+            lowest,
+          ) should have message s"requirement failed: highest ($highest) is out of range [0, ${VersionedOffsetBuilder.MaxHighest}]"
+      }
+    }
+
+    "fail on a negative middle index" in {
+      forAll(arbitrary[Byte], genHighest, Gen.negNum[Int], Gen.posNum[Int]) {
+        (version, highest, middle, lowest) =>
+          the[IllegalArgumentException] thrownBy VersionedOffsetBuilder(version).of(
+            highest,
+            middle,
+            lowest,
+          ) should have message s"requirement failed: middle ($middle) is lower than 0"
+      }
+    }
+
+    "fail on a negative lowest index" in {
+      forAll(arbitrary[Byte], genHighest, Gen.posNum[Int], Gen.negNum[Int]) {
+        (version, highest, middle, lowest) =>
+          the[IllegalArgumentException] thrownBy VersionedOffsetBuilder(version).of(
+            highest,
+            middle,
+            lowest,
+          ) should have message s"requirement failed: lowest ($lowest) is lower than 0"
+      }
+    }
+
+    "fail on a wrong version" in {
+      forAll(genHighest, Gen.posNum[Int], Gen.posNum[Int], genDifferentVersions) {
```

Comment:
Can you add test that we do check the offset's version for `VersionedOffsetBuilder.split` as well, please? That should cover the other setters/getters as well like `setLowestIndex`.

Reply:
Done